File: packages/demo/test/models/session.test.js



import { Workspace, reset } from './helpers.js';

beforeEach(reset);

test(`session`, () => Workspace.run(async _ => {
    const { sessions, users } = _.database;

    expect(await sessions.count()).toBe(0);

    const user = await users.insert({
        name: 'Admin',
        email: 'admin@example.com',
        role: 'admin'
    });

    const session = await sessions.insert({
        userId: user.id,
        passString: 'foo'
    });

    expect(await sessions.count()).toBe(1);

    expect(await session.user.name).toBe('Admin');
}));

File: projects/demo/test/models/session_test.rb

require "test_helper"

Class.new Minitest::Test do

    include Slick::Helpers

    def setup
        run_command('reset-database')
    end

    def test_session
        assert_equal 0, sessions.count

        user = users.insert name: 'Admin', email: 'admin@example.com', role: 'admin'

        session = sessions.insert user_id: user.id, pass_string: 'foo'

        assert_equal 1, sessions.count

        assert_equal 'Admin', session.user.name
    end

end

